sdk/src/main/java/nl/rabobank/gict/payments_savings/omnikassa_frontend/sdk/Endpoint.java

```diff
@@ -46,13 +46,37 @@ public final class Endpoint {
      * @param baseURL       this is the Url that points to the Rabobank API
      * @param signingKey    this is the key given by the Rabobank to sign all communication
      * @param tokenProvider this must be your own implementation of the tokenProvider, see developer-manual
-     * @deprecated This method is deprecated, use the {@link Endpoint#createInstance(Environment, String, TokenProvider)} instead.
      * @return new instance of Endpoint
      * Note: for backward compatibility do not remove this method!
+     * @deprecated This method is deprecated, use the {@link Endpoint#createInstance(Environment, String, TokenProvider)} instead.
      */
     @Deprecated
     public static Endpoint createInstance(String baseURL, byte[] signingKey, TokenProvider tokenProvider) {
         ApiConnector connector = new ApiConnector(baseURL, signingKey);
+        return createInstance(baseURL, signingKey, tokenProvider, null, null);
+    }
+
+    /**
+     * Use this method to retrieve an instance of the Endpont
+     *
+     * @param baseURL          this is the Url that points to the Rabobank API
+     * @param signingKey       this is the key given by the Rabobank to sign all communication
+     * @param tokenProvider    this must be your own implementation of the tokenProvider, see developer-manual
+     * @param userAgent        this is the User-Agent value you want to give your implementation
+     * @param partnerReference this can be filled with the partner reference, if applicable.
+     * @return new instance of Endpoint
+     * Note: for backward compatibility do not remove this method!
+     * @deprecated This method is deprecated, use the {@link Endpoint#createInstance(Environment, String, TokenProvider)} instead.
+     */
+    @Deprecated
+    public static Endpoint createInstance(String baseURL,
```

Comment:
Are we sure we need to add this to the deprecated factory method as well? I see our test shop uses this factory method. Can't it be migrated to the new factory method?

Reply:
No we can't change this one, then we cannot test locally anymore